text/0000-mergeable-rustdoc-cross-crate-info.md

```diff
@@ -0,0 +1,441 @@
+* With this proposal, there will be duplicate logic to read from both `doc.parts` files and rendered CCI.
+
+[@jsha proposes](https://github.com/rust-lang/rfcs/pull/3662#issuecomment-2184077829) unconditionally generating and reading from `crate-info.json`, with no appending to the rendered crate info.
+
+## Item links?
+
+Require users to pass `--extern-html-root-url` on all external dependencies (current), vs. add a new flag to facilitate missing docs links being generated across all user crates (should consider)
+
+Cross-crate links are an important consideration when merging documentation. rustdoc provides a flag, `--extern-html-root-url`, which can provide fine-grain control over the generated URL for documentation. We believe this is already sufficient for generating cross-crate links, and we will make no changes to the generation of item links. Example usage of this flag is in the Guide-level explanation.
+
+A proposal is to provide more options to facilitate cross-crate links. For example, we could add a flag that implies `--extern-html-root-url` and `--include-info-json` on all crates passed through `--extern`. It could be something like `--extern-html-root-url-all-user-crates`. User crates would be taken to mean the transitive dependencies of the current crate, excluding the standard library. This is complicated by things like <https://doc.rust-lang.org/cargo/reference/unstable.html#build-std>.
+
+## Reuse existing option?
+
+Create a new flag, `--merge` (proposed), vs. use existing option `no_emit_shared`
+
+There is a render option, `no_emit_shared`, which is used to conditionally generate the cross-crate information. It also controls the generation of static files, user CSS, etc.
+
+This option is not configurable from the command line, and appears to be enabled unless rustdoc is run in its example scraping mode.
+
+We could make it configurable from the command line, unconditionally generate `doc.parts`, and use it to gate the merging of CCI.
+
+We could also make it configurable from the command line, and use it to gate the generation of `doc.parts` and the generation of all of the shared files.
+
+We could also leave it as-is: always false unless we're scraping examples, and gate the generation of `doc.parts` and the generation of all of the shared files.
+
+# Future possibilities
+
+This change could begin to facilitate type aliases and the trait aliases being
+statically compiled as part of the .html documentation, instead of being loaded
+as separate JavaScript files. Each type and trait alias could be stored as an
+HTML part, which are then merged into the regular documentation.
+
+Another possibility is for `doc.parts` to be distributed on `docs.rs` along with the regular documentation. This would facilitate a mode where documentation of the dependencies could be downloaded externally, instead of being rebuilt locally.
```

Comment:
That would be very interesting!

Reply:
That is true. It would really only be useful if you wanted to avoid the additional work of documenting a crate locally, but the rmeta and the generated docs were available externally.

Reply:
My after-thought point here was that distributing these files on docs.rs would likely not bring any advantage (except for the search index maybe).